bin/entrypoint.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+# =================================================================
+#
+# Authors: Ricardo Garcia Silva <ricardo.garcia.silva@gmail.com>
+#
+# Copyright (c) 2017 Ricardo Garcia Silva
+#
+# Permission is hereby granted, free of charge, to any person
+# obtaining a copy of this software and associated documentation
+# files (the "Software"), to deal in the Software without
+# restriction, including without limitation the rights to use,
+# copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following
+# conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# =================================================================
+
+"""Entrypoint script for docker containers.
+
+This module serves as the entrypoint for docker containers. Its main
+purpose is to set up the pycsw database so that newly generated
+containers may be useful soon after being launched, without requiring
+additional input.
+
+"""
+
+
+import argparse
+import logging
+import os
+from six.moves.configparser import SafeConfigParser
+from subprocess import call
+from time import sleep
+
+from sqlalchemy import create_engine
+from sqlalchemy.exc import OperationalError
+from sqlalchemy.exc import ProgrammingError
+
+from pycsw.core import admin
+
+logger = logging.getLogger(__name__)
+
+
+def launch_pycsw(gunicorn_workers=2):
+    logger.debug("Reading pycsw config...")
+    config = SafeConfigParser()
+    config.read("/etc/pycsw/pycsw.cfg")
+    db_url = config.get("repository", "database")
```

Comment:
Should retrieve config from the environment instead of assuming it is still the default value